In [1]:
'''Packages Import'''
# Basic Data Handling
import numpy as np
import pandas as pd
# Data Visualization
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

# Statsmodels - Statistics

# Scikit-learn - Machine Learning

# Custom
from config import *
from util import *
from model import *

# Ignore Warnings
import warnings
warnings.filterwarnings('ignore')

Output Cleaning & Datasets Loading

In [3]:
delete_files(path=OUTPUT_PATH, exception=STATIC_FNAME)
df = download_file(url=DOWNLOAD_URL, unzip='Y', output=OUTPUT_PATH, dt_col=['Date'])
df.head(1)

Store       Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0      1 2010-05-02     1643690.9             0        42.31       2.572   

          CPI  Unemployment  
0  211.096358         8.106

Exploratory Data Analysis (EDA)

Basic Infor: shape, type & missing

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6435 entries, 0 to 6434
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   Store         6435 non-null   int64         
 1   Date          6435 non-null   datetime64[ns]
 2   Weekly_Sales  6435 non-null   float64       
 3   Holiday_Flag  6435 non-null   int64         
 4   Temperature   6435 non-null   float64       
 5   Fuel_Price    6435 non-null   float64       
 6   CPI           6435 non-null   float64       
 7   Unemployment  6435 non-null   float64       
dtypes: datetime64[ns](1), float64(5), int64(2)
memory usage: 402.3 KB


Based on the above:
1. 6435 records, with 8 columns;
2. No missing values; 
3. Data type looks reasonable.

Univariate Analysis

1. Categorical Variable

2. Continuous Variable

Handling of Mising & Extreme Values

Features' Distribution

Features' Relationship w/ Target

Feature Engineering

Baseline Modeling

Improvement

In [2]:
delete_files(path=OUTPUT_PATH, exception=STATIC_FNAME)

Future Work

    1. Use distributed system tools for data loading, such as PySpark